src/lib/utils/meta_table_manager.cpp

```diff
@@ -2,18 +2,84 @@
 
 #include "constant_mappings.hpp"
 #include "hyrise.hpp"
+#include "resolve_type.hpp"
 #include "statistics/table_statistics.hpp"
 #include "storage/base_encoded_segment.hpp"
+#include "storage/segment_iterables/any_segment_iterable.hpp"
 #include "storage/table.hpp"
 #include "storage/table_column_definition.hpp"
 
+namespace {
+
+using namespace opossum;  // NOLINT
+
+int32_t get_distinct_value_count(const std::shared_ptr<BaseSegment> segment) {
+  auto distinct_value_count = int32_t{0};
+  resolve_data_type(segment->data_type(), [&](auto type) {
```

Comment:
Should we have a specialized case for dictionary segments?

Reply:
Since it comes for free, I guess it makes sense. Will change.

Right now, I assume that a NULL is not a value. Is everybody onboard with this view? Hence, I would return the size of the dictionary.